tools/test/examples/update.py

```diff
@@ -232,106 +232,101 @@ def prepare_fork(arm_example):
             return False
     return True
 
-def prepare_branch(branch):
+def prepare_branch(src, dst):
     """ Set up at branch ready for use in updating examples 
         
     Description:
     
-    This function checks whether or not the supplied branch exists.
-    If it does not, the branch is created and pushed to the origin.
+    This function checks whether or not the supplied dst branch exists.
+    If it does not, the branch is created from the src and pushed to the origin.
     The branch is then switched to.
     
     Args:
-    arm_example - Full GitHub repo path for original example 
+    src - branch to create the dst branch from
+    dst - branch to update
+
+    Return:
     ret - True if the fork was synchronised successfully, False otherwise
     
     """
 
-    update_log.debug("Preparing branch: %s", branch)
+    update_log.debug("Preparing branch: %s", dst)
 
     # Check if branch already exists or not.
     cmd = ['git', 'branch']
     return_code, output = run_cmd_with_output(cmd)
 
-    if not branch in output:
+    if not dst in output:
+        
         # OOB branch does not exist thus create it and then check it out
-        cmd = ['git', 'checkout', '-b', branch]
+
+        # First ensure we are on the src branch
+        cmd = ['git', 'checkout', src]
         return_code = run_cmd(cmd)
         if not return_code:
-
-            # Push new branch upstream
-            cmd = ['git', 'push', '-u', 'origin', branch]
+                
+            cmd = ['git', 'checkout', '-b', dst]
             return_code = run_cmd(cmd)
+            if not return_code:
+
+                # Push new branch upstream
+                cmd = ['git', 'push', '-u', 'origin', dst]
+                return_code = run_cmd(cmd)
```

Comment:
Have improved it further, this function doesn't really need to have it's own separate error so can actually just use run_cmd(cmd, exit_on_failure=True). It wasn't even checking the return from prepare_fork() or prepare_branch() anyway!